src/android/Poynt.java

```diff
@@ -114,17 +230,232 @@ public void onActivityResult(int requestCode, int resultCode, final Intent data)
+                secondScreenService = IPoyntSecondScreenService.Stub.asInterface(service);
+                cbk.success("");
+            }
+        };
+        serviceConnectionI = new ServiceConnection() {
+            @Override
+            public void onServiceDisconnected(ComponentName name) {
+                businessService = null;
+                cbk.error("Error Binding businessService");
+                }
+
+            @Override
+            public void onServiceConnected(ComponentName name,
+                    IBinder service) {
+                businessService = IPoyntBusinessService.Stub.asInterface(service);
+                cbk.success("");
+            }
+        };
+        serviceConnectionB = new ServiceConnection() {
+            @Override
+            public void onServiceDisconnected(ComponentName name) {
+                mBillingService = null;
+                cbk.error("Error Binding BillingService");
+                }
+
+            @Override
+            public void onServiceConnected(ComponentName name,
+                    IBinder service) {
+                mBillingService = IPoyntInAppBillingService.Stub.asInterface(service);
+                cbk.success("");
+            }
+        };
+        cordova.getActivity().bindService(Intents.getComponentIntent(Intents.COMPONENT_POYNT_SECOND_SCREEN_SERVICE_V2),serviceConnection, BIND_AUTO_CREATE);
+        cordova.getActivity().bindService(Intents.getComponentIntent(Intents.COMPONENT_POYNT_BUSINESS_SERVICE),serviceConnectionI, BIND_AUTO_CREATE);
+        
+        Intent serviceIntent = new Intent("com.poynt.store.PoyntInAppBillingService.BIND");
+        serviceIntent.setPackage("com.poynt.store");
+        cordova.getActivity().bindService(serviceIntent,serviceConnectionB, BIND_AUTO_CREATE);
+        
+    }
+    
+     
+      
+   private void showConfirmation(String message) {
+        try {
+            Bundle options = new Bundle();
+            options.putString("FONT_COLOR", "#f07f22");
+            options.putString(Intents.EXTRA_CONTENT_TYPE, Intents.EXTRA_CONTENT_TYPE_HTML);
+            secondScreenService.displayMessage(message, options);
+        } catch (RemoteException e) {
+            e.printStackTrace();
+        }
+    }
+ 
+    public void showWelcome(String message) {
+        try {
+            Bundle options = new Bundle();
+            /*options.putString("FONT_COLOR", "#f07f22");
+            options.putString(Intents.EXTRA_CONTENT_TYPE, Intents.EXTRA_CONTENT_TYPE_HTML);
+            secondScreenService.displayMessage(message, options);*/
+            options.putString("FONT_COLOR", "#f07f22");
+            options.putString(Intents.EXTRA_CONTENT_TYPE, Intents.EXTRA_CONTENT_TYPE_HTML);
+            secondScreenService.displayMessage(message, options);
+        } catch (RemoteException e) {
+            e.printStackTrace();
+        }
+    }
+    
+    public void collectSignature(String tit, String msg, String butt) {
```

Comment:
please change to:
`public void collectSignature(String title, String msg, String button)
`
Instead of calling `showConfirmation` inside this method, it should be called from your javascript after your callback gets the signature.